### DISCUSSION TITLE
Unimplemented patterns

### DISCUSSION THREAD

**Discussion by 0xBradock** [rust-unofficial/patterns#389] [General]
Hello,

I got to this repo searching for patterns implementations.
I see that the umbrella issue points to some unimplemented issues like #232, #233 and #234 (I see there are others).
Is there any reason those are not yet implemented?
I would like to contribute.

There are also some patterns specific to Rust, that I think would be of great value to add.

**Comment by simonsan** [comment]
Sounds great! Which article for example would you be interested in writing as a starter? So I can guide you through the process of it.

I think the reason some of them are not implemented is one of the following:
- not really (easily) applicable to Rust
- time constraints of the maintainers to write own articles (as we are also having other projects)
- not a lot of interest of the community for this pattern, so nobody chimed in (yet) to write an article
(there are obviously more reasons, but these are probably the main ones)

**Comment by 0xBradock** [comment]
Awesome,

Well one pattern I am super familiar with and had implemented many times (for database connection) is the factory.
I saw the discussion on #224 

What about I start with this one?

**Reply by simonsan** [reply]
Factories are in so far a bit cumbersome, as Rust has `T::default`, which makes it super easy to instantiate a struct with a default implementation. So the indirection of e.g. having an enum, I personally don't know if it's interesting as a pattern for Rust. Also, I don't know of any examples in the wild.

Though, there are a few candidates, that are interesting to put into the book next (IMHO):
- https://github.com/rust-unofficial/patterns/issues/59
- https://github.com/rust-unofficial/patterns/pull/100
- Entry API: https://github.com/rust-unofficial/patterns/blob/e3ed5133461a3acd26e4ee44a0cdff9a494166e8/patterns/entry.md
- https://github.com/rust-unofficial/patterns/issues/220

If nothing from this list actually interests you, go with one that actually interests you most right now, or that you personally use most often. That's always the best approach to that, IMHO. So you have fun writing about something while having a lot of knowledge about something. :)

**Reply by 0xBradock** [reply]
That sounds great to me, thanks for the suggestions.
In that case I will be glad to go with the facade.

I read the instructions on the contributing guide and it is all clear for me.
So, if it is ok with you I'll try to have a draft PR by the end of the week.

**Reply by simonsan** [reply]
Sure, that sounds great! :)
